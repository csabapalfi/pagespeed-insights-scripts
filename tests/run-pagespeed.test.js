const {runPagespeed} = require('../lib/run-pagespeed');
const wreck = require('wreck');
jest.mock('wreck');

describe('run-pagespeed', () => {
  const lighthouseResult = {};

  it('runPageSpeed calls Mobile API and returns Lighthouse result', async () => {
    jest.spyOn(Date, 'now').mockImplementation(() => 1479427200000);
    wreck.get = jest.fn()
      .mockResolvedValue({payload: {lighthouseResult}});

    const {result} = await runPagespeed('https://www.google.com');

    expect(result).toBe(lighthouseResult);
    expect(wreck.get).toHaveBeenCalledTimes(1)
    expect(wreck.get.mock.calls[0]).toMatchSnapshot();
  });

  it('runPageSpeed calls Desktop API and returns Lighthouse result', async () => {
    jest.spyOn(Date, 'now').mockImplementation(() => 1479427200000);
    wreck.get = jest.fn()
      .mockResolvedValue({payload: {lighthouseResult}});

    const {result} = await runPagespeed('https://www.google.com', 'desktop');

    expect(result).toBe(lighthouseResult);
    expect(wreck.get).toHaveBeenCalledTimes(1)
    expect(wreck.get.mock.calls[0]).toMatchSnapshot();
  });

});
