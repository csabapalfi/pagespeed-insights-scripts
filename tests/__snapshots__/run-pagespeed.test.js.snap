// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`run-pagespeed runPageSpeed calls Desktop API and returns Lighthouse result 1`] = `
Array [
  "https://www.googleapis.com/pagespeedonline/v5/runPagespeed/?url=https%3A%2F%2Fwww.google.com%2F%237200000&strategy=desktop",
  Object {
    "json": true,
  },
]
`;

exports[`run-pagespeed runPageSpeed calls Mobile API and returns Lighthouse result 1`] = `
Array [
  "https://www.googleapis.com/pagespeedonline/v5/runPagespeed/?url=https%3A%2F%2Fwww.google.com%2F%237200000&strategy=mobile",
  Object {
    "json": true,
  },
]
`;
